lab4/viewpoints/context/context.tex

```diff
@@ -1,8 +1,349 @@
 \section{Context viewpoint}
 
+
+\subsection{System Scope and Responsibilities}
+
+\subsection*{Scope}
+EduPal is a learning aid platform for students and teachers. It allows educators to upload course information, and students to access learning material and tools.
+
+\begin{itemize}
+    \item Pomodoro Timer: Helps students manage study sessions effectively.
+    \item Note Creation and Export: Allows students to create notes and export them in PDF format.
+    \item Goal Setting: Enables users to set and track educational goals.
+    \item ChatGPT Integration: Provides an AI-powered assistant to help answer questions and provide learning support.
+    \item Subject and Topic Management: Users can organize their studies by subjects and topics, and upload conspectuses.
+    \item Real-Time Quizzing: This will allow users to create quiz rooms, share access codes, and participate in real-time quizzes, with results displayed immediately.
+\end{itemize}
+
+
+\subsection{Identity of External Entities and Services and Data Used}
+
+\subsection*{External Entities}
+\begin{itemize}
+    \item \textbf{Users (Students and Teachers):} Interact with EduPal to upload, manage, and access course content or take quizzes, utilise the features such as Pomodoro timer, quizzes, etc.
+    \item \textbf{PostgreSQL Database:} The central data storage system for EduPal. Stores all application data, except conspectus files.
```

Comment:
Database is part of the system (asked the lecturer) and is too low level to be shown on the diagram or talked about in the text. Must be removed. (Same with OpenAI API btw)
However, we will use object storage external system (for image storage) that must be in the text and diagrams (see my messenger message).

Reply:
There won't be any external entities then, except for the object storage system. And does the OpenAI API not fit here because it isn't related to quizzes?

Reply:
Indeed, only object storage system. I asked the lecturer and he said that we only need to put stuff related to the change we make.

Reply:
How does that system work, like a noSQL database?

Reply:
> How does that system work, like a noSQL database?

like a filesystem on steroids (S3, Azure blob storage). For context its enough to know that the system has the capabilities to store files and retrieve them.